qiskit/providers/ibmq/runtime/runtime_program.py

```diff
@@ -255,3 +258,77 @@ class ProgramResult(NamedTuple):
     name: str
     description: str
     type: str
+
+class ParameterNamespace(SimpleNamespace):
+    """ An abstraction for SimpleNamespace that offers param validation.
+
+    Note: Unused/Unnecessary params does not cause error (still validates)
+
+    Args:
+        params (List[ProgramParameter]): The program's input parameters
+
+    Raises
+        IBMQInputValueException on validation fail
+    """
+
+    def __init__(self, params: List[ProgramParameter]):
+        super().__init__()
+        # Allow access to the raw program parameters list
+        self.params_list = params
+        # For localized logic, create store of parameters in dictionary
+        self.__program_params: dict = {}
+
+        for param in params:
+            # (1) Add parameters to a dict by name
+            setattr(self, param.name, None)
+            # (2) Store the program params for validation
+            self.__program_params[param.name] = param
+
+    def validate(self) -> None:
+        """Validates the user's usage of the program's inputs
+
+        Raises:
+            IBMQInputValueError if validation fails
+        """
+
+        # Iterate through the user's stored inputs
+        for param_name, program_param in self.__program_params.items():
+            # Set invariants: User-specified parameter value (value) and whether it's required (req)
+            value = None
+            req = program_param.required
+            # Check there exists a program parameter of that name.
+            try:
+                # Get the value currently stored for this parameter (user-adjusted)
+                value = self.__dict__[param_name]
+            except AttributeError:
+                if req:
+                    raise IBMQInputValueError('Param (%s) missing required value!' % param_name)
+            # Check for empty value
+            if not value or value is None:
+                if req:
+                    raise IBMQInputValueError('Param (%s) missing required value!' % param_name)
+
+            value_type = type(value).__name__
+            # Check program param of that name is of same type
+            if value is not None and value_type != program_param.type:
+                raise IBMQInputValueError('Parameter (%s) is incorrect type! (%s != %s)' % \
+                                          (param_name, value_type, program_param.type))
```

Comment:
Types cannot be validated because it's currently only used for documentation and there are no rules on what can/cannot be specified - e.g. one may specify `string` instead of `str`.

Reply:
@jyu00 Will programs with incorrectly-specified params just fail at runtime then? 
If there's no validation on types, this could be a good spot to catch that before it pings the API with invalid info (might reduce number of calls). 
If aliases are an issue, there could be a simple (constant) lookup table for some common aliases (integer -> int, string -> str), but the program's input types are available for the user so they should be able to use the same type letter-for-letter.

Reply:
Yes they'll just fail at runtime. A lookup table won't work because right now the types are entirely arbitrary and up to the program writers. To have a lookup table we'd first need to enforce the types can only be those in the table, and we don't have that yet.

Reply:
Ok, not a problem!